src/snowflake/snowpark/mock/plan.py

```diff
@@ -1661,7 +1670,9 @@ def calculate_expression(
                     ).sf_type
                     if not calculated_sf_type:
                         calculated_sf_type = cur_windows_sf_type
-                    elif calculated_sf_type.datatype != cur_windows_sf_type.datatype:
+                    elif not is_datatype_compatible(
```

Comment:
This basically allows coercion to happen but doesn't set the result type properly?

Reply:
i removed it